src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,232 @@
+and by the Stone-Weierstrass theorem it is the unique star algebra equivalence which extends the
+polynomial functional calculus (i.e., `polynomial.aeval`).
+
+This file also includes a proof of the **spectral permanence** theorem for (unital) C⋆-algebras
+(see `star_subalgebra.spectrum_eq`)
+
+## Main definitions
+
+* `continuous_functional_calculus : C(spectrum ℂ a, ℂ) ≃⋆ₐ[ℂ] elemental_star_algebra ℂ a`
+* `elemental_star_algebra.character_space_homeo :
+  `character_space ℂ (elemental_star_algebra ℂ a) ≃ₜ spectrum ℂ a`
+
+## Main statements
+
+* `star_subalgebra.coe_is_unit`: for `x : S` in a C⋆-subalgebra `S` of `A`, then `↑x : A` is a unit
+  if and only if `x` is a unit.
+* `star_subalgebra.spectrum_eq`: **spectral_permanence** for `x : S`, where `S` is a C⋆-subalgebra
+  of `A`, `spectrum ℂ x = spectrum ℂ (x : A)`.
+
+## Notes
+
+The result we have established here is the strongest possible, but it is likely not the version
+which will be most useful in practice. Future work will include developing an appropriate API for
+the continuous functional calculus (including one for real-valued functions with real argument that
+applies to self-adjoint elements of the algebra). -/
+
+open_locale pointwise ennreal nnreal complex_order
+
+open weak_dual weak_dual.character_space elemental_star_algebra
+
+variables {A : Type*} [normed_ring A] [normed_algebra ℂ A]
+variables [star_ring A] [cstar_ring A] [star_module ℂ A]
+
+instance {R A : Type*} [comm_ring R] [star_ring R] [normed_ring A] [algebra R A] [star_ring A]
+  [has_continuous_star A] [star_module R A] (a : A) [is_star_normal a] :
+  normed_comm_ring (elemental_star_algebra R a) :=
+{ mul_comm := mul_comm, .. subring_class.to_normed_ring (elemental_star_algebra R a) }
+
+instance {R A : Type*} [normed_field R] [star_ring R] [normed_ring A] [normed_algebra R A]
+  [star_ring A] [has_continuous_star A] [star_module R A] (a : A) :
+  normed_algebra R (elemental_star_algebra R a) :=
+star_subalgebra.to_normed_algebra (elemental_star_algebra R a)
+
+instance {R A : Type*} [normed_field R] [star_ring R] [normed_ring A] [normed_algebra R A]
+  [star_ring A] [has_continuous_star A] [star_module R A] (a : A) :
+  normed_space R (elemental_star_algebra R a) :=
+normed_algebra.to_normed_space _
+
+-- without this instance type class search causes timeouts
+noncomputable instance elemental_star_algebra.complex.normed_algebra (a : A) :
+  normed_algebra ℂ (elemental_star_algebra ℂ a) :=
+infer_instance
+
+variables [complete_space A] (a : A) [is_star_normal a] (S : star_subalgebra ℂ A)
+
+/-- This lemma is used in the proof of `star_subalgebra.is_unit_of_is_unit_of_is_star_normal`,
+which in turn is the key to spectral permanence `star_subalgebra.spectrum_eq`, which is itself
+necessary for the continuous functional calculus. Using the continuous functional calculus, this
+lemma can be superseded by one that omits the `is_star_normal` hypothesis. -/
```

Comment:
Is it already possible to state and prove easily the better version? If so, it would be good to include it in this PR, because otherwise I am afraid we will forget to improve this docstring once it lands.

Reply:
It's not particularly simple, no. This is the hard direction in showing the equivalence between the different definitions of positivity (i.e., selfadjoint with nonnegative spectrum vs. has a selfadjoint square root vs. is the product of an element and its star). Moreover, we'll likely need to make several API decisions regarding the continuous functional calculus before we get there. I'll try to keep this docstring on my radar for the future.